plugins/plugin-chart-echarts/src/utils/series.ts

```diff
@@ -78,14 +83,22 @@ export function extractGroupbyLabel({
   groupby,
   numberFormatter,
   timeFormatter,
+  coltypes,
 }: {
   datum?: DataRecord;
   groupby?: string[] | null;
   numberFormatter?: NumberFormatter;
   timeFormatter?: TimeFormatter;
+  coltypes?: Array<GenericDataType>;
 }): string {
   return (groupby || [])
-    .map(val => formatSeriesName(datum[val], { numberFormatter, timeFormatter }))
+    .map((val, index) =>
+      formatSeriesName(datum[val], {
+        numberFormatter,
+        timeFormatter,
+        ...(coltypes && { coltype: coltypes[index] }),
```

Comment:
I'm not sure we can assume the index in `coltypes` corresponds to `groupby`; the result set can include other columns that aren't in groupby, like metrics, temporal columns etc. So to make sure this works properly, we should probably make a mapping from `coltypes` to `colnames` before using the types for formatting here.